### ISSUE TITLE
Domain WG Applicant: Machine learning

### ISSUE THREAD

**Issue by skade** [rust-lang/wg-governance#11]
This issue acts as a tracking issue and log for the application of a machine learning domain WG.

Initial can be found here:
https://internals.rust-lang.org/t/enabling-the-formation-of-new-working-groups/10218/11?

Current Status:
* [x] Application sent
* [ ] Charta written

The initial goal would be to find a pair of governance WG members to facilitate the formation of the group from our side.

**Comment by skade** [comment]
Personal opinion: Machine learning has been a high-hopes subject for Rust before and still is, adopting it as a focus for the wider project might give it some weight, even if competition is currently high.

**Comment by deg4uss3r** [comment]
Just commenting to say I am the proposer for the RustML WG, happy to hear any suggestions from the community!

**Comment by LukeMathWalker** [comment]
Long overdue!
Linking this here as a reference for pre-existing discussions on the topic: https://github.com/rust-ml/discussion/issues/1

**Comment by deg4uss3r** [comment]
@LukeMathWalker This is amazing, great work! Very happy to see the community behind this huge potential for Rust :)

**Comment by ThomAub** [comment]
I would be happy to help with the construction of this WG. As I'm still learning Rust I could first help with the organization.

**Comment by sherif7810** [comment]
I would like to help. I need to refresh my knowledge of Rust. I like it.

I suggest we create a deep learning library accelerated by TVM.
They have Rust bindings. Also, it is going to save us implementing automatic differentiation and hardware acceleration on various devices.
I think this is the most straight forward way and the lowest hanging fruit. All we have to do is create the best ergonomic and most flexible framework on top of it.

**Comment by mkubasz** [comment]
You have my keyboard :) I will be happy to create some good system to ML integrated with rustism and ndarray .

**Comment by unidentified9** [comment]
I would like to help as well. I've been wanting to learn rust, ML and more about common software practices. I'm not really all that experienced at any of this but I'm interested in learning and understanding more.

**Comment by ruchira** [comment]
Excited about this, hope to be able to help at some point.

**Comment by swarnimarun** [comment]
I have been curious about Rust and ML for a while now, really excited to see what awesome things can be brought up here. :)
Hopefully I can be of some help as well... Even though I am only a beginner.

**Comment by matbesancon** [comment]
I would be interested to follow along and help here, but I think the WG should be on scientific computing in general and not just on machine learning

**Comment by BatmanAoD** [comment]
@matbesancon I believe that we want to keep WGs fairly tightly focused; I don't know that a topic _broader_ than machine learning would be appropriate. I'd definitely like to hear what sort of things you would want such a WG to address, though; maybe those topics could be shaped into a separate WG proposal, or perhaps I'm mistaken and they'd be appropriate to fold into this proposal.

**Comment by matbesancon** [comment]
@BatmanAoD, I see your point. The thing is, most building blocks for scientific computing are the same in many areas, ML is not special in that regard. Having a working group making sure these building blocks are available and robust does not broaden the scope of the work to do pe se, but includes people working on scientific computing tasks which are not learning.

**Comment by thankrandomness** [comment]
I agree with @matbesancon. Working Group should focus on developing building blocks to enable a thriving scientific computing community in Rust. Machine Learning automatically falls under that.

**Comment by deg4uss3r** [comment]
@skade Was just curious where we were in the process? Happy to jump on the discord channel if it works better to talk there!

**Comment by XAMPPRocky** [comment]
@deg4uss3r I have emailed the core team, asking them to follow up about the application.

**Comment by deg4uss3r** [comment]
Thank you @XAMPPRocky you're awsome!

**Comment by xflowXen** [comment]
Just came across this and quite keen to be involved. Also agree with @matbesancon that the working group should focussed on scientific computing. Is anyone actively championing this at the minute - did the core team ever respond?

**Comment by deg4uss3r** [comment]
@xflowXen Yes! We are trying to get the core foundation of the WG down over at the preexisting[rust-ml](https://github.com/rust-ml/discussion/issues) group. I am the original proposer and currently championing the effort :)
